server/gitrest/packages/gitrest-base/src/utils/redisFs/redisFsManager.ts

```diff
@@ -0,0 +1,457 @@
+	): Promise<string | undefined>;
+	public async mkdir(
+		folderpath: PathLike,
+		options?: Mode | (MakeDirectoryOptions & { recursive?: false | undefined }) | null,
+	): Promise<void>;
+	public async mkdir(
+		path: PathLike,
+		options?: Mode | MakeDirectoryOptions | null,
+	): Promise<string | undefined>;
+	public async mkdir(
+		folderpath: PathLike,
+		options?: Mode | MakeDirectoryOptions | null,
+	): Promise<undefined | string | void> {
+		const folderpathString = folderpath.toString();
+
+		await executeRedisFsApi(
+			async () => this.redisFs.set(folderpath.toString(), ""),
+			RedisFsApis.Mkdir,
+			RedisFSConstants.RedisFsApi,
+			this.redisFsConfig.enableRedisFsMetrics,
+			this.redisFsConfig.redisApiMetricsSamplingPeriod,
+			{
+				folderpathString,
+			},
+		);
+	}
+
+	/**
+	 * Removes the directory identified by folderpath. For removing subdirectories and files,
+	 * `rm` should be used instead.
+	 * For more info on how isomorphic-git uses this, see:
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L152
+	 */
+	public async rmdir(folderpath: PathLike, options?: RmDirOptions): Promise<void> {
+		const keys_to_remove = await this.redisFs.keys_by_prefix(folderpath.toString());
+
+		keys_to_remove.forEach((key) => {
+			void this.redisFs.delete(key);
+		});
+	}
+
+	/**
+	 * Returns an fs.Stat object for a given filepath. fs.Stat provides information about
+	 * a file or folder.
+	 * `fs.Stats` would usually include information like file size and modified time. However, it is
+	 * very expensive to fetch that information from redis every time. Because we don't employ
+	 * fields like `stats.size` and `stats.mtime` in our specific use-case scenario of isomorphic-git,
+	 * we can avoid making "get properties" requests to Az storage, as the only other types of information
+	 * we need are whether a path exists and whether it refers to a file or directory. We can compute that
+	 * independently. (Please note that we are initially enabling this change for only a subset of tenants)
+	 * For more info on how isomorphic-git uses this, see:
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L61
+	 * https://github.com/isomorphic-git/isomorphic-git/blob/main/src/models/FileSystem.js#L191
+	 * Neither isomorphic-git nor GitRest currently use the options parameter.
+	 */
+	public async stat(
+		filepath: PathLike,
+		options?: StatOptions & { bigint?: false | undefined },
+	): Promise<Stats>;
+	public async stat(
+		filepath: PathLike,
+		options: StatOptions & { bigint: true },
+	): Promise<BigIntStats>;
+	public async stat(filepath: PathLike, options?: StatOptions): Promise<Stats | BigIntStats>;
+	public async stat(filepath: PathLike, options?: any): Promise<Stats | BigIntStats> {
+		const filepathString = filepath.toString();
+		const data = await executeRedisFsApi(
+			async () => this.redisFs.get<Buffer | string>(filepathString),
```

Comment:
If Isomorphic-Git calls `stat()` for a folder (which would be a key prefix), how would it work? I think the `mkdir` command is creating keys with empty values just to represent the fact that a directory exists, but I'm not sure if I missed something